### PR TITLE
Enhance braking

### PR DESCRIPTION
**Pull request by nyukhalov**
- Increase the range of detecting traffic lights
- Decreased deceleration when braking
- Changed the `decelerate_waypoints` method in order to make the car brake more actively

Here is the difference between velocity functions
https://www.desmos.com/calculator/07bo5linys

### CODE REVIEW

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import math
 
 STATE_COUNT_THRESHOLD = 3
-TL_LOOK_AHEAD = 50
+TL_LOOK_AHEAD = 100
```
Comment (elias3) [inline ros/src/tl_detector/tl_detector.py:19]: What's the impact of this change ?

Comment (nyukhalov) [reply ros/src/tl_detector/tl_detector.py:19]: I should have put a comment there..
This defines how far the car sees traffic lights.
If the distance between the car and a traffic light is too small, the car will suddenly find the traffic light, and will not have enough time to stop. Having too big distance results in extra work our tl_detector node should do.

**Review by elias3 — DISMISSED** [review]

File: ros/src/tl_detector/tl_detector.py (hunk 1 above)
Comment (kradio3) [reply ros/src/tl_detector/tl_detector.py]: I've also changed it to 100. So after rebase we see something like this:
```
- TL_LOOK_AHEAD =100 
+ TL_LOOK_AHEAD = 100
```

**Review by kradio3 — COMMENTED** [review]
Only one question: does TL still detects light, when car stays ahead of stop line?
There is a logic which prevents detection after stop line. Should i change it?

**Comment by nyukhalov** [comment]
There is an issue with control when a car gets a bit ahead of the closest stopline.
It's not a big issue because it only happens when the car drives quite fast.
We can change it in another PR, just to not block this change.

**Review by elias3 — APPROVED** [review]

**Comment by elias3** [comment]
Please add a story in trello so we can track it

**Comment by nyukhalov** [comment]
Sure, here is the card: https://trello.com/c/IO61dyBA/25-the-car-starts-driving-when-then-traffic-light-is-red

**Review by kradio3 — APPROVED** [review]

File: ros/src/twist_controller/twist_controller.py [hunk 2]
```diff
@@ -63,7 +63,9 @@ def control(self, dbw_enabled, current_vel, target_linear_vel, target_angular_ve
             brake = 700
         elif throttle < .1 and vel_error < 0:
             throttle = 0
-            decel = max(vel_error, self.decel_limit)
+            decel = 0.5 * vel_error
```
Comment (elias3) [inline ros/src/twist_controller/twist_controller.py:66]: Can you just add a small comment about the meaning of 0.5 here

**Pull request merged**